core/amber/src/main/scala/edu/uci/ics/texera/reporter/workFlowReporter.scala

```diff
@@ -0,0 +1,316 @@
+package edu.uci.ics.texera.reporter
+
+import com.typesafe.config.Config
+import edu.uci.ics.amber.engine.common.virtualidentity.ActorVirtualIdentity
+import edu.uci.ics.texera.reporter.workFlowReporter.Configuration
+import kamon.Kamon
+import kamon.metric.MeasurementUnit.Dimension.{Information, Time}
+import kamon.metric.MeasurementUnit.{information, time}
+import kamon.metric._
+import kamon.module.MetricReporter
+import kamon.tag.{Lookups, Tag, TagSet}
+import kamon.util.{EnvironmentTags, Filter}
+import org.slf4j.LoggerFactory
+
+import java.text.{DecimalFormat, DecimalFormatSymbols}
+import java.time.Duration
+import java.util.Locale
+import scala.collection.mutable
+import scala.collection.mutable.ListBuffer
+import scala.jdk.CollectionConverters.asScalaBufferConverter
+
+class workFlowReporter(
+                        @volatile private var configuration: Configuration,
```

Comment:
Why is this volatile?

Reply:
In Kamon project, the reporter configuration may be changed in different threads. And every reporter cycle when the reporter store its data(metrics), it will read the configuration for some variables. For example, I can change the unit, or actor filter. I think volatile can make the configuration more safe. But if you think that 'no volatile' is better, it is also fine.